corehq/apps/users/views/mobile/users.py

```diff
@@ -662,6 +663,9 @@ def post(self, request, *args, **kwargs):
                 device_id="Generated from HQ",
                 user_data=self.custom_data.get_data_to_save(),
             )
+            if 'location_id' in kwargs:
+                couch_user.set_location(Location.get(kwargs['location_id']))
```

Comment:
I think we want to be more protective if the location doesn't exist or is in the wrong domain. should probably use `get_document_or_404`